src/compiler/checker.ts

```diff
@@ -48585,6 +48637,28 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
         return false;
     }
+    function isLiteralComputedName(node: ComputedPropertyName) {

```

Comment:
This feels like it overlaps with `hasBindableName` a _lot_. If I had to tell you at a glance how this differed in its' selectivity from that, and why, I'm not sure I could. Should this be `hasBindableName(node.parent)` with some extra addenda further restricting it?

Reply:
You are right, `hasBindableName` does the same thing, the only check this function adds is the guard against type narrowing. 

```ts
       const symbol = getSymbolAtLocation(expression);
        if (!symbol) {
            return false;
        }
        // Ensure not type narrowing
        const declaredType = getTypeOfSymbol(symbol);
        return declaredType === type;        
```


Could we add a flag to `hasBindableName` maybe called `noNarrowing`? Or keep this function and use `hasBindableName` for the first part and just keep the guard against narrowing. I'll implement the second one for ow and we can discuss further.

Reply:
I'd say keep this function but use `hasBindableName` for the first part. That should at least ensure they stay in-sync.